### DISCUSSION TITLE
🥂 iLogtail开源社区线上Meetup 第1期（时间：8月3日晚19:30-20:30）

### DISCUSSION THREAD

**Discussion by messixukejia** [alibaba/ilogtail#251] [Announcements]
# 概要
议题：iLogtail开源社区线上Meetup
时间：8月3日晚19:30-20:30
直播地点：[iLogtail社区 钉钉群](https://github.com/iLogtail/ilogtail-docs/raw/main/.gitbook/assets/chatgroup.png)
参与人：[@messixukejia](https://github.com/messixukejia)、[@yyuuttaaoo](https://github.com/yyuuttaaoo)、[@EvanLjp](https://github.com/EvanLjp)

# 议题
* 如何进行iLogtail开发及基本的问题定位？[@yyuuttaaoo](https://github.com/yyuuttaaoo)
* [社区贡献流程](https://github.com/alibaba/ilogtail/blob/main/docs/cn/contributing/CONTRIBUTING.md)、[社区建设招募](https://github.com/alibaba/ilogtail/issues/279)（包括不限于Feature开发、UT、Bug修复、文档完善、案例分享等）。[@messixukejia](https://github.com/messixukejia)
* [开发者成长路线](https://github.com/alibaba/ilogtail/blob/main/docs/cn/contributing/developer.md)。[@messixukejia](https://github.com/messixukejia)
* iLogtail社区版答疑。现收集相关问题，我们将会对热点问题进行线上交流，欢迎留言。

<img src="https://user-images.githubusercontent.com/13250519/182148204-20f9c91f-ca73-4a18-b1b6-14f2f8dde865.png" height="50%" width="50%" >

<img src="https://user-images.githubusercontent.com/13250519/182506620-82cb6b6e-9e18-43f1-b26e-32935e788e0c.png" height="50%" width="50%" >

**Comment by sunym1991** [comment]
使用SLS好像可以进行全局的采集配置管控，那iLogtail开源是否有多节点采集配置管控能力？

**Reply by messixukejia** [reply]
SLS的全局管控是SLS服务端进行的，可以通过[机器组](https://help.aliyun.com/document_detail/89088.html)进行机器的管理；SLS控制台或API配置的采集配置，可以通过机器组发送到iLogtail侧生效。

iLogtail社区版不具备SLS服务端管控的能力，目前只能进行单机配置，或者借助Supervised等第三方工具进行管控，或者在K8s上利用configmap进行配置管理。目前我们也在规划开源版本的管控服务端，可以全局管理所有注册的iLogtail节点，后续这部分也会开源。

**Reply by sunym1991** [reply]
给你们点赞，期待后续的管控服务端开源

**Comment by Takuka0311** [comment]
iLogtail开源输出到各种存储系统支持比较少，未来是什么规划的？

**Reply by yyuuttaaoo** [reply]
iLogtail未来将通过flusher插件对形式支持更多的开源存储系统，目前对于呼声较高的Kafka、ES、CK都建立了对应issue进行追踪，在这里也呼吁一下有兴趣的小伙伴帮忙一起开发和优化对应插件的功能。

**Comment by messixukejia** [comment]
[logtail商业版和社区版的区别](https://github.com/alibaba/ilogtail/discussions/268)

**Comment by StartE** [comment]
想请教一下，iLogtail开源版官方文档写的是镜像里编译，对于Linux、WIndow、MAC本地是否支持？ 能否解答一下

**Reply by yyuuttaaoo** [reply]
对于Linux，目前是支持本地编译的，但编译需要必要的编译器和库。您根据这张清单构建依赖的库：https://ilogtail.gitbook.io/ilogtail-docs/installation/sources/dependencies 。您也可以尝试从编译镜像的/opt/logtail/deps目录docker cp编译依赖，但这些预编译的库与host的linux兼容性需要您自己确认。同样目前的ilogtail编译器为4.8.5，其他版本的编译器不保证一定可以兼容。
Windows和Mac当前仅支持本地编译Go语言写到插件，C++暂时不能本地编译。
我们期望iLogtail能在更多的环境上编译和运行，但目前我们的重点仍然在Linux上，如果你希望与我们一起完善多平台的支持，请不要犹豫加入我们。

**Reply by messixukejia** [reply]
企业版的iLogtail我们是支持[Linux（X86/ARM 64）](https://help.aliyun.com/document_detail/28982.html)、[Windows](https://help.aliyun.com/document_detail/49006.html)的。

**Comment by JeffPhoneix** [comment]
社区版本的配置和阿里云官网页面上的logtail配置是完全相同的吗？文档是互通的吗？还是说其实是两套，不要混在一起？

**Reply by messixukejia** [reply]
本质可以认为是相同的。阿里云官网页面上的logtail 使用json配置，因为历史兼容原因有些配置比较复杂，层次不是太清晰，但是因为更多的在用控制台操作，所以也不会解决太复杂，但是如果直接手动配置就比较复杂了。

因此开源版做了一些简化的处理，提供了新的结构层次更清晰的yaml配置。其实yaml配置最后还是转换成json生效的，只是对于使用者不可见。

原理基本可以认为是相通的，配置项略有差异，基本保持了一致性。

比如说processor_rename，虽然一个json，一个yaml，但是总体上是相同的，最大差异就是yaml配置去掉了detail这一层级。
企业版配置：https://help.aliyun.com/document_detail/196157.html

```
{
  "processors":[
    {
      "type":"processor_rename",
      "detail": {
        "SourceKeys": ["aaa1","aaa2"],
        "DestKeys": ["bbb1","bbb2"]
      }
    }
  ]
}
```

社区版：https://ilogtail.gitbook.io/ilogtail-docs/data-pipeline/processor/processor-rename

```processors:
  - Type: processor_rename
    SourceKeys: 
      - key1
    DestKeys:
      - new_key1
```

**Reply by messixukejia** [reply]
之前经常看到会提一些关于ilogtail的问题，是否方便留一下联系方式？我们便于了解你的情况，以便进一步的合作。https://github.com/alibaba/ilogtail/issues/277  麻烦提供下钉钉号 或 邮箱，这样我们可以联系到你。

**Comment by yyuuttaaoo** [comment]
回放视频 https://www.bilibili.com/video/BV1zU4y1e7Rk/?spm_id_from=333.999.0.0
